cmd/release/docs/internal/component_table.go

```diff
@@ -38,39 +44,107 @@ func GetComponentVersionsTable(release *Release) (string, error) {
+	captureRegexForLine := regexp.MustCompile(fmt.Sprintf(`uri: (%s.*)`, ecrBase))
+	foundMatches := captureRegexForLine.FindAllSubmatch(body, -1)
+
+	// Assumes uri format is "<ecrBase>/[some_stuff]/<NAME>:v<VERSION>-<uriSuffix>", where NAME and VERSION are targeted.
+	// Example: "public.ecr.aws/eks-distro/etcd-io/etcd:v3.4.15-eks-1-20-2" -> NAME is "etcd" and VERSION is "3.4.15"
 	captureRegexForName := regexp.MustCompile(`.*/(.*):v`)
-	captureRegexForVersion := regexp.MustCompile(`:v(.*)-eks`)
+	captureRegexForVersion := regexp.MustCompile(fmt.Sprintf(`:v(.*)-%s`, uriSuffix))
 
 	var tableRows []string
 
-	//	Example uri value and what it creates from this:
-	//		uri: public.ecr.aws/eks-distro/etcd-io/etcd:v3.4.14-eks-1-18-5
-	//		output: | etcd | 3.4.14 | public.ecr.aws/eks-distro/etcd-io/etcd:v3.4.14-eks-1-18-5 |
+	// Example uri value and what it creates from this:
+	//		input uri from match:	public.ecr.aws/eks-distro/etcd-io/etcd:v3.4.15-eks-1-20-2
+	//		output table row:		| etcd | 3.4.15 | public.ecr.aws/eks-distro/etcd-io/etcd:v3.4.15-eks-1-20-2 |
 	for _, matchPair := range foundMatches {
 		uri := matchPair[1]
-		name := captureRegexForName.FindSubmatch(uri)[1]
-		version := captureRegexForVersion.FindSubmatch(uri)[1]
+		name := string(captureRegexForName.FindSubmatch(uri)[1])
+		version := string(captureRegexForVersion.FindSubmatch(uri)[1])
+		tableRows = append(tableRows, formatComponentRow(name, version, string(uri)))
+	}
 
-		tableRow := fmt.Sprintf("| %s | %s | %s |", name, version, uri)
-		tableRows = append(tableRows, tableRow)
+	assetsVersion, _ := GetKubernetesReleaseGitTag(release.Branch())
+	for _, assetName := range assetsNotInReleaseManifest {
+		uri := fmt.Sprintf("%s/kubernetes/%s:%s-%s", ecrBase, assetName, assetsVersion, uriSuffix)
+		tableRows = append(tableRows, formatComponentRow(assetName, strings.TrimPrefix(assetsVersion, "v"), uri))
 	}
 
-	otherAssetsVersion, _ := GetKubernetesReleaseGitTag(release.Branch())
-	for _, asset := range assetsNotInReleaseManifest {
-		uri := fmt.Sprintf("%s/kubernetes/%s:%s-%s", ecrBase, asset, otherAssetsVersion, release.EKSBranchNumber)
-		tableRow := fmt.Sprintf("| %s | %s | %s |", asset, otherAssetsVersion[1:], uri) // [1:] removes 'v'
+	sort.Strings(tableRows)
+
+	tableRows = append([]string{tableHeader, tableHeaderBodyDivider}, tableRows...)
+
+	return strings.Join(tableRows, linebreakString), nil
+}
 
-		tableRows = append(tableRows, tableRow)
+// GetComponentVersionsTableWithSameComponentVersionsAsPrevRelease returns Markdown table of presumed component versions
+// for the release.
+// The previous patch release's component version table must exist in the expected file and in the expected table format.
+// Returned table uses the same components and their versions as the previous patch release. The returned table is not
+// checked for accuracy or compared with the release manifest. References to the EKS-D release number are updated.
```

Comment:
I get that this is being done for the purpose of the bot, but why can it not be compared with the release manifest for accuracy? What is our confidence going with this route then?